src/coord/src/coord.rs

```diff
@@ -327,9 +328,11 @@ pub struct Coordinator {
     /// Tracks write frontiers for active exactly-once sinks.
     sink_writes: HashMap<GlobalId, SinkWrites<Timestamp>>,
 
-    /// A map from pending peeks to the queue into which responses are sent, and
-    /// the IDs of workers who have responded.
-    pending_peeks: HashMap<u32, mpsc::UnboundedSender<PeekResponse>>,
+    /// A map from pending peek ids to the queue into which responses are sent, and
+    /// the connection id of the client that initiated the peek.
+    pending_peeks: HashMap<Uuid, (mpsc::UnboundedSender<PeekResponse>, u32)>,
```

Comment:
This is becoming a @benesch trope, but can you extract a `struct PendingPeek { sender, conn_id }` here? Just makes the code way more self documenting.